### PR TITLE
Return correct Content-Type from an AddMetricsHandler wrapped handler

### PR DESCRIPTION
**Pull request by nullseed**
Signed-off-by: Alex Young <alex@heuris.io>

From testing master after #346 was merged: after the reader handler is wrapped in the metrics handler, headers such as Content-Type are not forwarded. That means that GET /system/functions always returns text/html even if the reader returns application/json. This can be observed in postman by issuing the above request. This is existing behaviour.

This PR addresses the issue above and adds unit tests that describe the changes.

Note that add_metrics_test.go is in the metrics package so that coverage of the add_metrics.go statements could be seen.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Alex you're certainly making your mark here. Thank you for the attention to detail.

File: gateway/metrics/add_metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+type testFetchQuery struct {
+	res *VectorQueryResponse
+	err error
+}
+
+func (f testFetchQuery) Fetch(query string) (*VectorQueryResponse, error) {
+	return f.res, f.err
+}
+
+func TestAddMetricsHandlerServeHTTPReturnsContentTypeOfPassedInHandlerWhenFetchSucceeds(t *testing.T) {
```
Comment (alexellis) [inline gateway/metrics/add_metrics_test.go]: I am not clear what these tests are doing. We always need to return Content-Type of `application/json`.

Comment (nullseed) [reply gateway/metrics/add_metrics_test.go]: These tests prove the issue described in the PR. They can be extended to cover more of the logic in the AddMetricsHandler function.

Comment (alexellis) [reply gateway/metrics/add_metrics_test.go]: I think the function needs to return `application/json` the whole time, unless it's an error as in an exception.

**Comment by alexellis** [comment]
I think one of the Pos that went in recently addressed the "null" response. Would you like to verify whether this is still necessary?

**Comment by alexellis** [comment]
Closing as I'm not sure if this is necessary anymore. Have pinged 4 days ago.

**Pull request closed without merging**